src/libraries/System.Formats.Tar/src/System/Formats/Tar/SubReadStream.cs

```diff
@@ -0,0 +1,180 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Formats.Tar
+{
+    // Stream that allows wrapping a super stream and specify the lower and upper limits that can be read from it.
+    // It is meant to be used when the super stream is unseekable.
+    // Does not support writing.
+    internal class SubReadStream : Stream
```

Comment:
How does this code differ from https://github.com/dotnet/runtime/blob/accec1c4d4bc0a923b7cfa189d9cf65a943054dc/src/libraries/System.IO.Compression/src/System/IO/Compression/ZipCustomStreams.cs#L221?  I'm wondering why we can't share the code.

Reply:
I made many changes and I didn't want to alter the behavior of the Compression one.